File: PlatziCodingChallenge_Dia44/SumaRecursiva.js
// Suma Recursiva en JavaScript

function sumaRecursiva(suma = 0) {
    const numero = parseFloat(prompt('Introduzca un número (0 o nada para terminar)'))
    if (numero === 0 || isNaN(numero)) {
        return suma
    } else {
        suma += numero
        return sumaRecursiva(suma)
    }
}

const suma = sumaRecursiva()
alert(`La suma de todos sus números es ${suma}`)

File: Dia44/SumaRecursiva.py
# Suma Recursiva en Python

def sumaRecursiva(suma = 0):
    entrada = input('Introduzca un número (0 o nada para terminar): ')
    try:
        numero = float(entrada)
        if (numero == 0):
            return suma
        else:
            suma += numero
            return sumaRecursiva(suma)
    except:
        return suma

suma = sumaRecursiva()
print("La suma de todos sus números es {}".format(suma))